### DISCUSSION TITLE
Running a different version of the data importer

### DISCUSSION THREAD

**Discussion by BerghsJelmer** [alexbelgium/hassio-addons#390] [Q&A] [answered]
Hi

i used the dataimporter with your addon. it ran into an issue which i fixed on a different branch on the original firefly dataimporter repo. Can you help me run my branch on the addon? or any way to run my branch with homeassistant?

**Comment by alexbelgium** [comment, marked as answer]
Hi, that's not straightforward. First, you should fork the repo, modify the config.json of the addon to remove reference to the image tag (it will then autobuild) then change the build tag in dockerfile to your own prebuilt container. You then add the forked address in your HA and install the updated addon. Other option : modify on the fly the files you needed to modify with a script running at the start of the addon, while keeping the initial container. If you do that, I could include it in the main addon so that it only triggers if the problematic code is present (and it will stop trigerring if an official container of data importer is published with the fix).

**Reply by BerghsJelmer** [reply]
That second options seems like the easiest option for me now. its a small code fix so it shouldnt be hard to change on the fly when starting the addon. Can you explain more how i can add such a script?

**Reply by BerghsJelmer** [reply]
Ive already made a simple bash command to edit the correct file. i just need it to be ran after the container is setup. I think i could place it at the end of the entrypoint.sh file?

**Reply by alexbelgium** [reply]
That would be great. The best would be to add the script in the same level than https://github.com/alexbelgium/hassio-addons/blob/master/fireflyiii_data_importer/rootfs/etc/cont-init.d/99-run.sh

For example you could make a script named 98-fix.sh containing your code or something like that. Could you please make a push request with the modification? (I suppose you are using sed or a tool like that?) the only other thing you need to change is increment the version tag in config.json (for example add -2 to the current version) to launch the build process! 

Thanks!

**Reply by BerghsJelmer** [reply]
I created a pull request, is that fine?

**Reply by BerghsJelmer** [reply]
indeed i used sed
And i also update the version number in config.json in the PR

**Comment by alexbelgium** [comment]
Thanks, I'll merge it

**Reply by BerghsJelmer** [reply]
Thanks! how can i update it with this version on homeassistant now?

**Comment by alexbelgium** [comment]
The build is done. So just wait for the supervisor to refresh (a couple hours) or you can force refresh it from the same location where you added my custom repo to your list in the addons store

**Reply by BerghsJelmer** [reply]
Aah alright thanks for the quick help!
